### PR TITLE
t.rast.aggregate.seasons: new addon for GRASS 8

### PR DESCRIPTION
**Pull request by lucadelu**

### CODE REVIEW

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <h2>DESCRIPTION</h2>
 <em><b>t.rast.seasons</b></em> aggregates an input space time
```
Comment (veroandreo) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]:
```suggestion
<em><b>t.rast.aggregate.seasons</b></em> aggregates an input space time
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html [hunk 2]
```diff
@@ -12,14 +13,14 @@ <h2>DESCRIPTION</h2>
 <p>
 
 <h2>EXAMPLES</h2>
-Calculate seasonal data from
+Calculate seasonal data from an input space time raster datasets with an output space time raster datasets
```
Comment (veroandreo) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]:
```suggestion
Calculate seasonal data from an input space time raster dataset
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+# %option G_OPT_STRDS_OUTPUT
+# % label: The name of a singular space time raster dataset
+# % description: Using this option all the yearly space time raster datasets will be merged in a singular space time raster dataset
+# % required: no
+# %end
+
+# %option
+# % key: years
+# % type: string
+# % label: List of years, separator could be comma (list) or minus (range)
```
Comment (veroandreo) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]:
```suggestion
# % label: List of years, separator may be comma (list) or minus (range)
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+# % multiple: yes
+# % required: no
+# %end
+
+# %option
+# % key: basename
+# % type: string
+# % label: Basename of the new generated output maps and space time raster datasets
```
Comment (veroandreo) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]:
```suggestion
# % label: Basename of the new generated output maps
```

**Review by veroandreo — COMMENTED** [review]
minor grammar suggestions

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html (hunk 2 above)
Comment (lucadelu) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]: Probably the better change is "datasets with an output space time raster datasets" -> "dataset with a unified output space time raster dataset"

Comment (veroandreo) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]: yes, I like that. But then the manual should explain that if the output option is not used, the result is not a strds. Is that correct?

Comment (lucadelu) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]: No, right now without `output` option it create a singular space time raster dataset for each year of the input space time raster dataset. I'm going to improve documentation.

File: src/temporal/t.rast.aggregate.seasons/testsuite/test_seasons.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import grass.temporal as tgis
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+
+class TestClimatologies(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Initiate the temporal GIS and set the region"""
+        tgis.init(True)  # Raise on error instead of exit(1)
```
Comment (wenzeslaus) [inline src/temporal/t.rast.aggregate.seasons/testsuite/test_seasons.py]: Using a named (keyword) argument may remove the need for a comment:

```suggestion
        tgis.init(raise_fatal_error=True)
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+        if count % 10 == 0:
+            gs.percent(count, len(years), 1)
+
+    # Wait for unfinished processes
+    process_queue.wait()
+
+    if len(outputs) > 1:
+        pymod.Module("t.merge", inputs=",".join(outputs), output=output_name)
+
+    return True
```
Comment (wenzeslaus) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]: The return value is unused.

```suggestion
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+    # create t.rast.aggregate.ds module to be copied
+    mod = pymod.Module("t.rast.aggregate.ds")
+    mod.inputs.input = strds
+    mod.inputs.method = method
+    mod.inputs.basename = basename
+    mod.inputs.type = "stvds"
+    mod.flags.quiet = False
+    mod.flags.n = register_null
+    mod.flags.overwrite = gs.overwrite()
```
Comment (wenzeslaus) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]: The overwrite is transferred automatically. With this line, the behavior should be the same.

```suggestion
```

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+        temp_season = f"sample_seasons_{year}"
+        outsp = tgis.open_new_stds(
+            temp_season,
+            "stvds",
+            "absolute",
+            f"Season vector year {year}",
+            f"Season vector for the year {year}",
+            "mean",
+            dbif,
+            gs.overwrite(),
```
Comment (wenzeslaus) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]: I would expect that the global overwrite would be respected here, but I'm not sure from top of my head.

```suggestion
```

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py [hunk 9]
```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python
+
+
+################################################
+#
+# MODULE:       t.rast.seasons
+# AUTHOR(S):    Luca Delucchi
+# PURPOSE:      Aggregates an input strds with astronomical seasons granularity.
+#
+# COPYRIGHT:    (C) 2018 by Luca Delucchi
```
Comment (wenzeslaus) [inline src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py:10]: Not a big deal, but you may want to review copyright years and names of the tool at different places.

**Review by wenzeslaus — CHANGES_REQUESTED** [review]

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py (hunk 9 above)
Comment (lucadelu) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py:10]: Fixed in the new pull request

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html (hunk 2 above)
Comment (lucadelu) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.html]: Fixed in the new pull request #1010

File: src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py (hunk 4 above)
Comment (lucadelu) [reply src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py]: Fixed in #1010

**Comment by lucadelu** [comment]
I did some error with rebase, new pull request is #1010

**Pull request closed without merging**